### PR TITLE
Make sar run on all deployed machines, not only on Barman

### PR DESCRIPTION
**Pull request by barthisrael**
TPA had a `sys/sysstat` role which was in charge of configuring `sar` to run every minute, and enabling it.

However, the implementation would rely on `cron` service to be running, which was only started in nodes with the `barman` role.

Through this commit we amend that behavior in such a way `sar` will run on all nodes, not only on nodes that have the `barman` role.

We took advantage of that change to also modify how `sysstat` is configured in the servers: instead of using cronjobs, we will now use `systemd` timers. This is the approach that has been adopted for `sysstat` package on newer distros like Debian 11 and RHEL 9, so we changed TPA code to also do that.

In summary, this is what TPA will do on all supported distros:

* Create a `sysstat.service` unit, which is used as an entry point for booting `sa1` and installing `sysstat-collect.timer` and `sysstat-summary.timer` units
* Create `sysstat-collect.service` unit, which takes care of executing one round of 1 minute of `sa1`
* Create `sysstat-collect.timer` unit, which takes care of triggering the corresponding service every minute
* Create `sysstat-summary.service` unit, which performs an aggregation of metrics through `sa2`
* Create `sysstat-summary.timer` unit, which takes care of triggering the corresponding service every day at 7 o'clock

A few things needed to be taken into consideration:

* Debian/Ubuntu have the `sar` binaries under `/usr/lib/sysstat`
* RHEL/SLES have the `sar` binaries under `/usr/lib64/sa`
* `sa1` binary is named `debian-sa1` on Debian 10 and Ubuntu Focal
* When `sysstat` package is installed on Debian 10, Ubuntu Focal, or RHEL 7, the `sysstat.service` unit already comes enabled by default, so we need to disable it before enabling and starting it through `systemd` Ansible module, because attempting to enable an unit that is already enabled won't work as expected

References: TPA-595.

### CODE REVIEW

File: roles/sys/sysstat/tasks/main.yml [hunk 1]
```diff
@@ -2,4 +2,88 @@
...
+    # enable it
+    - name: Disable original sysstat service on Debian <= 10, Ubuntu <= 20, EL <= 7
+      systemd:
+        name: sysstat
+        enabled: false
+        daemon_reload: true
+      when: >
+        ansible_distribution == 'Debian' and ansible_distribution_major_version|int <= 10 or
+        ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int <= 20 or
+        ansible_os_family == 'RedHat' and ansible_distribution_major_version|int <= 7
```
Comment (amenonsen) [inline roles/sys/sysstat/tasks/main.yml:67]: Won't this task run on every deploy and keep disabling/enabling the service?

**Comment by amenonsen** [comment]
Looks generally fine, apart from the need to make sure we don't keep toggling the service on subsequent deploys.

File: roles/sys/sysstat/tasks/main.yml (hunk 1 above)
Comment (barthisrael) [reply roles/sys/sysstat/tasks/main.yml]: Makes sense, I'm submitting a commit to address that issue. Thanks!

File: roles/sys/sysstat/tasks/main.yml [hunk 2]
```diff
@@ -2,4 +2,99 @@
...
 # © Copyright EnterpriseDB UK Limited 2015-2023 - All rights reserved.
 
-- include_tasks: "os/{{ ansible_os_family }}.yml"
+- name: Set sar binaries path for Debian <= 10 or Ubuntu <= 20
+  set_fact:
+    sysstat_sa1_bin: /usr/lib/sysstat/debian-sa1
+    sysstat_sa2_bin: /usr/lib/sysstat/sa2
+  when: >
+    ansible_distribution == 'Debian' and ansible_distribution_major_version|int <= 10 or
+    ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int <= 20
```
Comment (haroon-github) [inline roles/sys/sysstat/tasks/main.yml:11]:
```suggestion
  when: >
    (ansible_distribution == 'Debian' and ansible_distribution_major_version|int <= 10) or
    (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int <= 20)
```

File: roles/sys/sysstat/tasks/main.yml [hunk 3]
```diff
@@ -2,4 +2,99 @@
...
+
+- name: Set sar binaries path for Debian > 10 or Ubuntu > 20
+  set_fact:
+    sysstat_sa1_bin: /usr/lib/sysstat/sa1
+    sysstat_sa2_bin: /usr/lib/sysstat/sa2
+  when: >
+    ansible_distribution == 'Debian' and ansible_distribution_major_version|int > 10 or
+    ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int > 20
```
Comment (haroon-github) [inline roles/sys/sysstat/tasks/main.yml:19]:
```suggestion
    (ansible_distribution == 'Debian' and ansible_distribution_major_version|int > 10) or
    (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int > 20)
```

File: roles/sys/sysstat/tasks/main.yml [hunk 4]
```diff
@@ -2,4 +2,99 @@
...
+      systemd:
+        name: sysstat
+        enabled: false
+        daemon_reload: true
+      when: >
+        sar_sysstat_collect_enabled.rc == 1 and (
+          ansible_distribution == 'Debian' and ansible_distribution_major_version|int <= 10 or
+          ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int <= 20 or
+          ansible_os_family == 'RedHat' and ansible_distribution_major_version|int <= 7
+        )
```
Comment (haroon-github) [inline roles/sys/sysstat/tasks/main.yml:78]:
```suggestion
        sar_sysstat_collect_enabled.rc == 1 and (
          (ansible_distribution == 'Debian' and ansible_distribution_major_version|int <= 10) or
          (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int <= 20) or
          (ansible_os_family == 'RedHat' and ansible_distribution_major_version|int <= 7)
        )
```

Comment (barthisrael) [reply roles/sys/sysstat/tasks/main.yml:78]: Sorry, I didn't notice there were more of these, otherwise I would've corrected these too.

**Review by haroon-github — APPROVED** [review]

**Pull request merged**